src/consumer/worker.js

```diff
@@ -24,10 +25,23 @@ const createWorker = ({ handler, workerId }) => {
       if (!item) break
 
       const { batch, resolve, reject } = item
+      const { topic, partition } = batch
+      const key = `${topic}|${partition}`
+      if (partitionAssignments.has(key)) {
+        logger.info('Skipping batch due to partition already being assigned to another worker', {
+          assignedWorker: partitionAssignments[partition],
+          topic,
+          partition,
+        })
+        continue
+      }
+      partitionAssignments.set(key, workerId)
```

Comment:
A batch can be in the queue, while it's not being processed. 
In that case, a fetcher can still add a duplicate batch to the queue. 

I would suggest keeping the partition assignments in fetch manager, but do something similar in fetcher instead - i.e. assign partitions from fetch response to nodeId(?), and for any fetch request, filter out partitions that are not in progress/queue already.

Reply:
Right, I'll give this a crack and see if I can make it work.

Reply:
Moved it up to the fetch manager/fetcher now, to make sure that only one of them can push batches for a topic-partition at any given time. I removed the filtering from the worker, as I don't believe it should be necessary anymore.